src/definitions/x509/validation/extensions/mod.rs

```diff
@@ -0,0 +1,260 @@
+        .any(|ki| {
+            issuer_skis.clone().any(|ski| {
+                tracing::debug!("comparing key identifiers:\n\t{ki:?}\n\t{:?}", ski.0);
+                ki == ski.0
+            })
+        })
+}
+
+/// Validate IACA extensions according to 18013-5 Annex B.
+pub fn validate_iaca_extensions(certificate: &Certificate) -> Vec<Error> {
+    tracing::debug!("validating IACA extensions...");
+
+    let extensions = certificate.tbs_certificate.extensions.iter().flatten();
+
+    let mut errors: Vec<Error> = check_for_disallowed_x509_extensions(extensions.clone());
+
+    errors.extend(
+        ExtensionValidators::default()
+            .with(SubjectKeyIdentifierValidator::from_certificate(certificate))
+            .with(KeyUsageValidator::iaca())
+            .with(BasicConstraintsValidator)
+            .with(CrlDistributionPointsValidator)
+            .with(IssuerAlternativeNameValidator)
+            .validate_extensions(extensions),
+    );
+
+    errors
+}
+
+/// Validate document signer extensions according to 18013-5 Annex B.
+pub fn validate_document_signer_certificate_extensions(certificate: &Certificate) -> Vec<Error> {
+    tracing::debug!("validating document signer certificate extensions...");
+
+    let extensions = certificate.tbs_certificate.extensions.iter().flatten();
+
+    let mut errors: Vec<Error> = check_for_disallowed_x509_extensions(extensions.clone());
+
+    errors.extend(
+        ExtensionValidators::default()
+            .with(SubjectKeyIdentifierValidator::from_certificate(certificate))
+            .with(ExtendedKeyUsageValidator {
+                expected_oid: document_signer_extended_key_usage_oid(),
+            })
+            .with(KeyUsageValidator::document_signer())
+            .with(CrlDistributionPointsValidator)
+            .with(IssuerAlternativeNameValidator)
+            .validate_extensions(extensions),
+    );
+
+    errors
+}
+
+/// Validate mdoc reader extensions according to 18013-5 Annex B.
+pub fn validate_mdoc_reader_certificate_extensions(certificate: &Certificate) -> Vec<Error> {
+    tracing::debug!("validating mdoc_reader certificate extensions...");
+
+    let extensions = certificate.tbs_certificate.extensions.iter().flatten();
+
+    let mut errors: Vec<Error> = check_for_disallowed_x509_extensions(extensions.clone());
+
+    errors.extend(
+        ExtensionValidators::default()
+            .with(SubjectKeyIdentifierValidator::from_certificate(certificate))
+            .with(ExtendedKeyUsageValidator {
+                expected_oid: mdoc_reader_extended_key_usage_oid(),
+            })
+            .with(KeyUsageValidator::mdoc_reader())
+            .with(CrlDistributionPointsValidator)
+            .with(IssuerAlternativeNameValidator)
+            .validate_extensions(extensions),
+    );
+
+    errors
+}
+
+#[derive(Default)]
+struct ExtensionValidators(Vec<Box<dyn ExtensionValidator>>);
+
+impl ExtensionValidators {
```

Comment:
You are implementing ExtensionValidators on line 177 as well. These methods could be defined together